### ISSUE TITLE
crash when opening IPython terminal from GUI

### ISSUE THREAD

**Issue by grlee77** [napari/napari#1473]
## 🐛 Bug

Clicking the "Open IPython console" icon at the bottom left of the GUI crashes the program when using `ipykernel` 5.1.1-5.1.4. The `setup.cfg` states that 5.1.1 is the minimum supported version, but the actual requirement seems to be >= 5.2.0.

<details><summary>Traceback</summary>

```
WARNING: Traceback (most recent call last):
  File "/media/lee8rx/data/git/3rdparty/napari/napari/_qt/qt_viewer.py", line 473, in toggle_console_visibility
    _ = self.console
  File "/media/lee8rx/data/git/3rdparty/napari/napari/_qt/qt_viewer.py", line 217, in console
    self.console = QtConsole({'viewer': self.viewer})
  File "/media/lee8rx/data/git/3rdparty/napari/napari/_qt/qt_console.py", line 80, in __init__
    kernel_client.start_channels()
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/qtconsole/kernel_mixins.py", line 47, in start_channels
    super(QtKernelClientMixin, self).start_channels(*args, **kw)
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/ipykernel/inprocess/client.py", line 64, in start_channels
    super(InProcessKernelClient, self).start_channels()
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/jupyter_client/client.py", line 118, in start_channels
    self.control_channel.start()
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/jupyter_client/client.py", line 200, in control_channel
    socket, self.session, self.ioloop
TypeError: ChannelABC() takes no arguments
18:39:14 WARNING Traceback (most recent call last):
  File "/media/lee8rx/data/git/3rdparty/napari/napari/_qt/qt_viewer.py", line 473, in toggle_console_visibility
    _ = self.console
  File "/media/lee8rx/data/git/3rdparty/napari/napari/_qt/qt_viewer.py", line 217, in console
    self.console = QtConsole({'viewer': self.viewer})
  File "/media/lee8rx/data/git/3rdparty/napari/napari/_qt/qt_console.py", line 80, in __init__
    kernel_client.start_channels()
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/qtconsole/kernel_mixins.py", line 47, in start_channels
    super(QtKernelClientMixin, self).start_channels(*args, **kw)
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/ipykernel/inprocess/client.py", line 64, in start_channels
    super(InProcessKernelClient, self).start_channels()
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/jupyter_client/client.py", line 118, in start_channels
    self.control_channel.start()
  File "/home/lee8rx/miniconda3/envs/pyir/lib/python3.7/site-packages/jupyter_client/client.py", line 200, in control_channel
    socket, self.session, self.ioloop
TypeError: ChannelABC() takes no arguments
```

</details>

## To Reproduce

1. install napari with ipykernel 5.1.x
```
conda create -y -n napari-env python=3.8
conda activate napari-env
pip install napari[all]
pip install "ipykernel==5.1.4"
```
2.  start `napari` from the command line
3.  click the terminal icon

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

```
napari: 0.3.5
Platform: Linux-4.15.0-108-generic-x86_64-with-glibc2.10
Python: 3.8.3 | packaged by conda-forge | (default, Jun 1 2020, 17:43:00) [GCC 7.5.0]
Qt: 5.14.2
PyQt5: 5.14.2
NumPy: 1.19.0
SciPy: 1.5.1
Dask: 2.20.0
VisPy: 0.6.4

GL version: 4.6.0 NVIDIA 440.100
MAX_TEXTURE_SIZE: 32768

Plugins:
- napari-plugin-engine: 0.1.6
- svg: 0.1.3
```

 - Any other relevant information:

bumping ipykernel to >=5.2.0 resolves the problem.

**Comment by sofroniewn** [comment]
Thanks for flagging this!! We've had some problems with that dependency in the past. I know we failed on `5.1.2`, but thought we were fixed for `5.1.4` (which it looks like I'm on now too). Looks like ipykernel is now on `5.3.2` which I just upgraded too and everything works well.

We could either update to `>5.2.0` or `>5.3.2` curious @jni if you have a preference (putting aside that we might hopefully soon make the napari console dependency optional)

[Edit: fix 5.1.14 -> 5.1.4 and 5.1.12 -> 5.1.2]

**Comment by grlee77** [comment]
>  know we failed on 5.1.12, but thought we were fixed for 5.1.14 (which it looks like I'm on now too)

You mean 5.1.4, not 5.1.14? (I see [5.1.4 as the most recent 5.1.x release](https://github.com/ipython/ipykernel/releases)).

**Comment by jni** [comment]
@sofroniewn I'd rather blacklist bad versions than bump our dependency to something super recent. But it seems that you and @grlee77 were both on 5.1.4? In which case maybe this is platform-dependent?

**Comment by sofroniewn** [comment]
This was closed by #1617 when we added a test with minimum version numbers to our test matrix and we were forced to update to `>5.2.0`. Thanks for the report!!
